### PR TITLE
Fixes #19969: Don't allow directory traversal through plugin URL

### PR DESCRIPTION
**Pull request by amousset**
https://issues.rudder.io/issues/19969

### CODE REVIEW

File: relay/sources/rudder-pkg/lib/rudder-pkg/rudderPkg.py [hunk 1]
```diff
@@ -388,9 +388,9 @@ def update():
     update_licenses()
 
 
-"""
+'''
```
Comment (Fdall) [inline relay/sources/rudder-pkg/lib/rudder-pkg/rudderPkg.py:391]: This looks like an outdated `blue`. They changed their policy on quotes in August.

Comment (amousset) [reply relay/sources/rudder-pkg/lib/rudder-pkg/rudderPkg.py:391]: oh

Comment (amousset) [reply relay/sources/rudder-pkg/lib/rudder-pkg/rudderPkg.py]: indeed.

**Comment by amousset** [comment]
PR updated with a new commit

File: relay/sources/rudder-pkg/lib/rudder-pkg/rudderPkgUtils.py [hunk 2]
```diff
@@ -214,18 +215,22 @@ def check_url():
...
 
 
 def download(completeUrl, dst=''):
     if dst == '':
-        fileDst = FOLDER_PATH + '/' + completeUrl.replace(URL + '/' + RUDDER_MINOR + '/', '')
+        # only take file name to avoid reaching max file name length
+        fileName = completeUrl.rsplit('/', 1)[-1]
+        # escape but keep tildes
+        fileName = urllib.parse.quote(fileName, '~')
+        fileDst = FOLDER_PATH + '/' + fileName
```
Comment (Fdall) [inline relay/sources/rudder-pkg/lib/rudder-pkg/rudderPkgUtils.py:233]: Looks good, can you force the `rpkg.index` download path in `rudderPkg.py` using something like:
`utils.download(utils.URL + '/' + 'rpkg.index', utils.INDEX_PATH)`

**Comment by amousset** [comment]
PR updated with a new commit

**Review by Fdall — APPROVED** [review]

**Comment by amousset** [comment]
OK, squash merging this PR

**Pull request merged**